presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -672,7 +673,11 @@ public PlanOptimizers(
                 ruleStats,
                 statsCalculator,
                 costCalculator,
-                ImmutableSet.of(new PushDeleteIntoConnector(metadata)))); // Must run before AddExchanges
+                ImmutableSet.of(
+                        // Must run before AddExchanges
+                        new PushDeleteIntoConnector(metadata),
+                        // Must run before AddExchanges and after join reordering
+                        new OptimizeOutputDuplicatesInsensitiveJoinRule(metadata))));
```

Comment:
I've added comment here:
```
// Must run after join reordering because join reordering creates
// new join nodes without JoinNode.maySkipOutputDuplicates flag set
```